restrict-validity-period/src/main.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see
+// <http://www.gnu.org/licenses/>.
+
+use chrono::{DateTime, FixedOffset, NaiveDate};
+use log::info;
+use slog::{slog_o, Drain};
+use slog_async::OverflowStrategy;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use transit_model::{Model, Result};
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "restrict-validity-period",
+    about = "Restrict the validity period of a NTFS and purge out-of-date data.",
+    rename_all = "kebab-case"
+)]
+struct Opt {
+    /// input directory.
+    #[structopt(short, long, parse(from_os_str), default_value = ".")]
+    input: PathBuf,
+
+    /// start of the desired validity period [included], e.g. 2019-01-01
+    #[structopt(short, long)]
+    start_validity_date: NaiveDate,
+
+    /// end of the desired validity period [included], e.g. 2019-01-01
+    #[structopt(short, long)]
+    end_validity_date: NaiveDate,
+
+    /// output directory
+    #[structopt(short, long, parse(from_os_str))]
+    output: PathBuf,
+
+    /// current datetime
+    #[structopt(
+        short = "x",
+        long,
+        parse(try_from_str),
+        default_value = &transit_model::CURRENT_DATETIME
+    )]
+    current_datetime: DateTime<FixedOffset>,
+}
+
+fn init_logger() -> slog_scope::GlobalLoggerGuard {
+    let decorator = slog_term::TermDecorator::new().stdout().build();
+    let drain = slog_term::CompactFormat::new(decorator).build().fuse();
+    let mut builder = slog_envlogger::LogBuilder::new(drain).filter(None, slog::FilterLevel::Info);
+    if let Ok(s) = std::env::var("RUST_LOG") {
+        builder = builder.parse(&s);
+    }
+    let drain = slog_async::Async::new(builder.build())
+        .chan_size(256) // Double the default size
+        .overflow_strategy(OverflowStrategy::Block)
+        .build()
+        .fuse();
+    let logger = slog::Logger::root(drain, slog_o!());
+
+    let scope_guard = slog_scope::set_global_logger(logger);
+    slog_stdlog::init().unwrap();
+    scope_guard
+}
+
+fn run(opt: Opt) -> Result<()> {
+    info!("Launching restrict-validity-period...");
+
+    let model = transit_model::ntfs::read(opt.input)?;
+    let mut collections = model.into_collections();
+    collections.restrict_period(opt.start_validity_date, opt.end_validity_date)?;
+    collections.sanitize()?;
```

Comment:
You can remove `collections.sanitize()?;` normally. It is performed after (in Model::new)